tests/helics/shared_library/iterationTests.cpp

```diff
@@ -1,268 +1,317 @@
-#include <boost/test/unit_test.hpp>
 #include <complex>
+#include <future>
+#include <gtest/gtest.h>
+#include <iostream>
 
-/** these test cases test out the value converters
- */
-#include "helics/application_api/Publications.hpp"
-#include "helics/application_api/Subscriptions.hpp"
-#include "helics/application_api/ValueConverter.hpp"
-
-namespace utf = boost::unit_test;
-
-BOOST_FIXTURE_TEST_SUITE(iteration_tests, FederateTestFixture, *utf::label("ci"))
+struct iteration_tests: public FederateTestFixture, public ::testing::Test {
+};
 
 /** just a check that in the simple case we do actually get the time back we requested*/
-#if ENABLE_TEST_TIMEOUTS > 0
-BOOST_TEST_DECORATOR(*utf::timeout(5))
-#endif
-BOOST_AUTO_TEST_CASE(execution_iteration_test)
+
+TEST_F(iteration_tests, execution_iteration_test)
 {
-    SetupTest<helics::ValueFederate>("test", 1);
-    auto vFed1 = GetFederateAs<helics::ValueFederate>(0);
+    SetupTest(helicsCreateValueFederate, "test", 1);
+    auto vFed1 = GetFederateAt(0);
     // register the publications
-    auto pubid = vFed1->registerGlobalPublication<double>("pub1");
+    auto pubid = helicsFederateRegisterGlobalPublication(
+        vFed1, "pub1", helics_data_type_double, "", nullptr);
 
-    auto subid = vFed1->registerRequiredSubscription<double>("pub1");
-    vFed1->setTimeDelta(1.0);
-    vFed1->enterInitializationState();
-    vFed1->publish(pubid, 27.0);
+    auto subid = helicsFederateRegisterSubscription(vFed1, "pub1", "", nullptr);
+    helicsFederateSetTimeProperty(vFed1, helics_property_time_delta, 1.0, nullptr);
 
-    auto comp = vFed1->enterExecutionState(helics::helics_iteration_request::iterate_if_needed);
+    helicsFederateEnterInitializingMode(vFed1, nullptr);
+    helicsPublicationPublishDouble(pubid, 27.0, nullptr);
 
-    BOOST_CHECK(comp == helics::iteration_result::iterating);
-    auto val = vFed1->getValue<double>(subid);
-    BOOST_CHECK_EQUAL(val, 27.0);
+    auto comp = helicsFederateEnterExecutingModeIterative(
+        vFed1, helics_iteration_request_iterate_if_needed, nullptr);
+    EXPECT_TRUE(comp == helics_iteration_result_iterating);
+    auto val = helicsInputGetDouble(subid, nullptr);
+    EXPECT_EQ(val, 27.0);
 
-    comp = vFed1->enterExecutionState(helics::helics_iteration_request::iterate_if_needed);
+    comp = helicsFederateEnterExecutingModeIterative(
+        vFed1, helics_iteration_request_iterate_if_needed, nullptr);
 
-    BOOST_CHECK(comp == helics::iteration_result::next_step);
+    EXPECT_TRUE(comp == helics_iteration_result_next_step);
 
-    auto val2 = vFed1->getValue<double>(subid);
+    auto val2 = helicsInputGetDouble(subid, nullptr);
 
-    BOOST_CHECK_EQUAL(val2, val);
+    EXPECT_EQ(val2, val);
 }
 
-#if ENABLE_TEST_TIMEOUTS > 0
-BOOST_TEST_DECORATOR(*utf::timeout(5))
-#endif
-BOOST_AUTO_TEST_CASE(execution_iteration_test_2fed)
+std::pair<double, int> runInitIterations(helics_federate vfed, int index, int total)
```

Comment:
It looks like this function keeps iterating until all of the publications change by a value < .001?